File: frontend/src/features/form/types/form-response-dto.ts
import { QuestionResponseDto } from "@/features/form/types/question-response-dto";

export enum FormStatus {
  Public = "PUBLIC",
  Unpublic = "UNPUBLIC",
  Private = "PRIVATE",
  Draft = "DRAFT",
  Closed = "CLOSED",
}

export enum Language {
  En = "EN",
  Pl = "PL",
}

export enum ShuffleVariant {
  Null = "NULL",
  Questions = "QUESTIONS",
  Answers = "ANSWERS",
  Both = "BOTH",
}

export interface FormResponseDto {
  id: string;
  name: string;
  slug: string;
  description?: string;
  language: Language;
  status: FormStatus;
  shuffleVariant: ShuffleVariant;
  thanksMessage: string;
  estimatedDuration: number;
  thumbnailKey: string;
  allowsQuestionPreview: boolean;
  allowsGuestSubmissions: boolean;
  saveSubmissions: boolean;
  authorId: string;
  createdAt: string;
  updatedAt: string;
  submissionCount: number;
  questions: QuestionResponseDto[];
}

File: backend/src/main/java/format/backend/form/application/shared/dto/FormResponseDto.java
package format.backend.form.application.shared.dto;

import static com.fasterxml.jackson.annotation.JsonFormat.Shape.NUMBER_FLOAT;

import com.fasterxml.jackson.annotation.JsonFormat;
import format.backend.form.domain.entity.FormLanguage;
import format.backend.form.domain.entity.FormShuffleVariant;
import format.backend.form.domain.entity.FormStatus;
import java.time.Duration;
import java.time.Instant;
import java.util.List;
import org.jspecify.annotations.Nullable;

public record FormResponseDto(
        String id,

        String name,

        String slug,

        @Nullable String description,

        FormLanguage language,

        FormStatus status,

        @Nullable FormShuffleVariant shuffleVariant,

        @Nullable String thanksMessage,

        Duration estimatedDuration,

        @Nullable String thumbnail,

        boolean allowsQuestionsPreview,

        boolean allowsGuestSubmissions,

        boolean saveSubmissions,

        boolean showAnswersFeedback,

        List<QuestionResponseDto> questions,

        long submissionsCount,

        long ratingsCount,

        @JsonFormat(shape = NUMBER_FLOAT, pattern = "0.0") @Nullable Double ratingAvg,

        @Nullable Integer userRating,

        @Nullable String authorName,

        Instant createdAt,

        Instant updatedAt) {}
